newIDE/app/src/ObjectsList/index.js

```diff
@@ -431,6 +439,15 @@ const ObjectsList = React.forwardRef<Props, ObjectsListInterface>(
       [onObjectCreated, objectsContainer]
     );
 
+    const swapObjectAsset = React.useCallback((object: gdObject) => {
+      setObjectAssetSwappingDialogOpen({ object });
+    }, []);
+
+    const canSwapAssetOfObject = React.useCallback((object: gdObject) => {
+      const type = object.getType();
+      return type === 'Scene3D::Model3DObject' || type === 'Sprite';
+    }, []);
```

Comment:
I think this can be moved to a function exported in newIDE/app/src/AssetStore/AssetSwappingDialog.js, so the check for type is colocated with the dialog doing the swap.

(and so there will be no need for useCallback)